### PR TITLE
Common changes

### PR DESCRIPTION
**Pull request by sjaeckel**
# Description

This PR contains the common (non board-specific) changes done while working on the port for Xilinx Versal.

# Testing

Ran `make check`.

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

**Comment by JacobBarthelmeh** [comment]
ok to test

File: wolfcrypt/test/test.c [hunk 1]
```diff
@@ -682,7 +682,7 @@ THREAD_RETURN WOLFSSL_THREAD wolfcrypt_test(void* args)
 int wolfcrypt_test(void* args)
 #endif
 {
-    int ret;
+    int ret = 0, errors_are_fatal = 0;
```
Comment (JacobBarthelmeh) [inline wolfcrypt/test/test.c:685]: How does the errors_are_fatal variable get set?

**Review by JacobBarthelmeh — CHANGES_REQUESTED** [review]

File: wolfcrypt/test/test.c [hunk 2]
```diff
@@ -803,660 +803,393 @@ options: [-s max_relative_stack_bytes] [-m max_relative_heap_memory_bytes]\n\
 #endif
 #endif
 
+#define TEST_RUN(fn, name) do {                       \
+    int ret_ = fn();                                  \
+    if (ret_ != 0 && errors_are_fatal) {              \
+        return err_sys(name "test failed!\n", ret_);  \
+    } else if (ret_ != 0) {                           \
+        printf(name " test failed (%d)!\n", ret_);    \
```
[the remark this answers is not in the document]
Comment (sjaeckel) [reply wolfcrypt/test/test.c:811]: old behavior should be restored

File: wolfcrypt/test/test.c (hunk 1 above)
Comment (sjaeckel) [reply wolfcrypt/test/test.c:685]: Manually by editing the source code :)

that's the only place where it's set ... could also be a define or `static const int`.

File: wolfcrypt/benchmark/benchmark.c [hunk 3]
```diff
@@ -7682,7 +7712,7 @@ int main(int argc, char** argv)
             base2 = 0;
 #if defined(HAVE_AESGCM) || defined(HAVE_AESCCM)
         else if (string_matches(argv[1], "-no_aad"))
-            aesAuthAddSz = 0;
+            aes_aad_options = 0x2U;
```
[the remark this answers is not in the document]
Comment (sjaeckel) [reply wolfcrypt/benchmark/benchmark.c:7716]: the code handling the `aes_aad_options` does this now

File: wolfcrypt/benchmark/benchmark.c [hunk 4]
```diff
@@ -2909,20 +2935,22 @@ static void bench_aesgcm_internal(int useDeviceID, const byte* key, word32 keySz
 
 void bench_aesgcm(int useDeviceID)
 {
+#define AES_GCM_STRING(n, dir)		AES_AAD_STRING("AES-" #n "-GCM-" #dir)
 #if defined(WOLFSSL_AES_128) && !defined(WOLFSSL_AFALG_XILINX_AES) \
-        && !defined(WOLFSSL_XILINX_CRYPT)
+        && (!defined(WOLFSSL_XILINX_CRYPT) || defined(versal))
```
[the remark this answers is not in the document]
Comment (sjaeckel) [reply wolfcrypt/benchmark/benchmark.c:2941]: This change sneaked in here, fixed now.

File: wolfcrypt/test/test.c (hunk 1 above)
Comment (sjaeckel) [reply wolfcrypt/test/test.c:685]: I've changed it to be modifiable by either defining `WOLFSSL_CRYPT_TEST_CONTINUE_ON_ERROR` on compile time and for the interactive tests (with arguments) by passing the `-c` option.

**Review by JacobBarthelmeh — CHANGES_REQUESTED** [review]
Looks good, the only thing left is the FIPS Windows failure. Which is due to private key read failure.

Steps to reproduce:

```
Download wolfssl-5.3.0-gplv3-fips-ready.zip  from www.wolfssl.com/download/
update wolfssl-5.3.0-gplv3-fips-ready/wolfcrypt/test/test.c with the test.c from this PR
open wolfssl-5.3.0-gplv3-fips-ready/IDE/WIN10/wolfssl-fips.sln in VS
build and run the project 'test'
update the core hash, line 1310 in fips_test.c
rebuild and run the project 'test'
```

**Comment by sjaeckel** [comment]
> Steps to reproduce:
> 
> ```
> Download wolfssl-5.3.0-gplv3-fips-ready.zip  from www.wolfssl.com/download/
> update wolfssl-5.3.0-gplv3-fips-ready/wolfcrypt/test/test.c with the test.c from this PR
> open wolfssl-5.3.0-gplv3-fips-ready/IDE/WIN10/wolfssl-fips.sln in VS
> build and run the project 'test'
> update the core hash, line 1310 in fips_test.c
> rebuild and run the project 'test'
> ```

exists an option to update the core hash w/o having to use VS on Windows?

**Pull request closed without merging**